### PR TITLE
feat: enable django flags support

### PR DESCRIPTION
**Pull request by pb82**
##### SUMMARY

Enable support for [django flags](https://cfpb.github.io/django-flags/).

Verification steps:
1. Start AWX locally
2. Start a bash shell in the container: `docker exec -it tools_awx_1 /bin/bash`
3. Set a test flag: `python manage.py enable_flag TEST_FLAG`
4. Add the following code to one of the views (e.g. the `ping` view):
```
from flags.state import flag_state
print(flat_state('TEST_FLAG'))

```
5. ensure that the value `True` is printed when accessing the route for the view
6. Disable the flag: `python manage.py disable_flag TEST_FLAG`
7. ensure that the value `False` is printed

##### ISSUE TYPE

 - New or Enhanced Feature

##### COMPONENT NAME

 - API

##### AWX VERSION

```
devel
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

File: requirements/requirements.txt [hunk 1]
```diff
@@ -23,7 +23,6 @@ asgiref==3.7.2
     #   channels-redis
     #   daphne
     #   django
-    #   django-ansible-base
```
Comment (AlanCoding) [inline requirements/requirements.txt]: These are still dependencies of DAB. They were added somewhat recently.

https://github.com/ansible/django-ansible-base/blob/devel/requirements/requirements_resource_registry.in

You probably generated this output incorrectly by using an outdated container image.

I also don't understand why we would add a dependency before we make any use of this dependency.

Comment (pb82) [reply requirements/requirements.txt]: @AlanCoding I made sure to pull the latest changes and rebuild the container image. But after running `./updater run` I still end up with the `django-ansible-base` comments removed.

File: requirements/requirements.txt [hunk 2]
```diff
@@ -422,7 +423,6 @@ requests==2.31.0
     #   -r /awx_devel/requirements/requirements.in
     #   adal
     #   azure-core
-    #   django-ansible-base
```
Comment (AlanCoding) [inline requirements/requirements.txt]: I don't want to bikeshed on this too much, but requests is still a requirement of DAB resource_registry, which is an AWX requirement.

https://github.com/ansible/django-ansible-base/blob/devel/requirements/requirements_resource_registry.in

So yes, it is unexpected that this got removed. But I can't say why.

Comment (pb82) [reply requirements/requirements.txt]: I'll see if I can find the reason for that.

Comment (pb82) [reply requirements/requirements.txt]: @AlanCoding  https://github.com/ansible/awx/pull/15677 has resolved it :+1:

**Review by AlanCoding — APPROVED** [review]

**Pull request merged**